subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -930,13 +946,29 @@ int nrf_cloud_rest_agps_data_get(struct nrf_cloud_rest_context *const rest_ctx,
 		goto clean_up;
 	}
 
+/** In filtered ephemeris mode, request A-GPS data no more often than
+ *  every 2 hours (time in milliseconds).
+ */
+#define AGPS_UPDATE_PERIOD (2 * 60 * 60 * MSEC_PER_SEC)
+
+	if (filtered && (last_request_timestamp != 0) &&
+	    (k_uptime_get() - last_request_timestamp) < AGPS_UPDATE_PERIOD) {
+		LOG_WRN("A-GPS request was sent less than 2 hours ago");
+		ret = 0;
+		result->agps_sz = 0;
+		goto clean_up;
+	}
+
 	init_request(&http_req, HTTP_GET, CONTENT_TYPE_APP_OCT_STR);
 
 	/* Determine size of URL buffer and allocate */
 	url_sz = sizeof(API_GET_AGPS_BASE);
 	if (request->net_info) {
 		url_sz += strlen(AGPS_NET_INFO) + NET_INFO_PRINT_SZ;
 	}
+	if (filtered) {
+		url_sz += strlen(AGPS_FILTERED) + strlen(AGPS_ELEVATION_MASK);
```

Comment:
is there any range checking on NRF_CLOUD_AGPS_ELEVATION_MASK?
a value of 3 digits or more would cause an error below?

Reply:
Good point.

Reply:
It's actually supposed to be limited to the range of 0 to 90 degrees.